### PR TITLE
feat: POC test utils iterator that prepends an index to each snapshot

### PR DESCRIPTION
**Pull request by colincclark**
- added `test-utils` package
- added iterator function to ensure snapshots are enumerated properly to ensure readable diffs
- added POC example to `article-flag`

NOTE: will add README once approved

### CODE REVIEW

File: packages/article-flag/__tests__/web/__snapshots__/article-flag-content.web.test.js.snap [hunk 1]
```diff
@@ -91,11 +91,11 @@ exports[`2. Red Article flag 1`] = `
     />
   </div>
   <div
-    aria-label="Coloured Red Flag"
+    aria-label="No Colour Flag"
```
Comment (craigbilner) [inline packages/article-flag/__tests__/web/__snapshots__/article-flag-content.web.test.js.snap]: should be red flag test?

File: packages/test-utils/__tests__/shared.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    name: "text example two",
+    test: () => exampleTest("second")
+  },
+  {
+    name: "ALL TEST NAMES SHOULD BE LOWERCASE",
+    test: () => exampleTest("third")
+  }
+];
+
+export default () => {
```
Comment (craigbilner) [inline packages/test-utils/__tests__/shared.js]: is no braces not a lint rule?

File: packages/test-utils/package.json [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    "eslint-config-prettier": "2.6.0",
+    "eslint-plugin-import": "2.8.0",
+    "eslint-plugin-jsx-a11y": "6.0.2",
+    "eslint-plugin-react": "7.4.0",
+    "jest": "21.2.1",
+    "prettier": "1.8.2",
+    "react-test-renderer": "16.3.1"
+  },
+  "dependencies": {
+    "@storybook/addon-actions": "3.4.1",
```
Comment (craigbilner) [inline packages/test-utils/package.json]: a lot of these deps look redundant?

File: packages/article-flag/__tests__/web/__snapshots__/article-flag-content.web.test.js.snap (hunk 1 above)
Comment (colincclark) [reply packages/article-flag/__tests__/web/__snapshots__/article-flag-content.web.test.js.snap]: done

File: packages/test-utils/src/index.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+/* global it */
+const iterator = tests =>
+  tests.map(({ name, test }, indx) => {
```
Comment (craigbilner) [inline packages/test-utils/src/index.js]: this returns a list of `true`, should it just be `forEach`?

File: packages/test-utils/package.json (hunk 3 above)
Comment (colincclark) [reply packages/test-utils/package.json]: done

File: packages/test-utils/__tests__/shared.js (hunk 2 above)
Comment (colincclark) [reply packages/test-utils/__tests__/shared.js]: it passed linting which is weird, but I removed the unrequired braces

https://github.com/newsuk/times-components/issues/1099

File: packages/test-utils/src/index.js (hunk 4 above)
Comment (colincclark) [reply packages/test-utils/src/index.js]: done

File: packages/test-utils/package.json [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  "version": "0.0.1",
+  "description": "Test helpers for Times Components",
+  "main": "dist/index",
+  "dev": "src/index",
+  "scripts": {
+    "fmt": "prettier --write '**/*.*'",
+    "prettier:diff": "prettier --list-different '**/*.*'",
+    "lint": "eslint . && yarn prettier:diff",
+    "test:android": "jest --config='./__tests__/android/jest.config.js'",
+    "test:ios": "jest --config='./__tests__/ios/jest.config.js'",
```
[the remark this answers is not in the document]
Comment (colincclark) [reply packages/test-utils/package.json]: thought the snapshots provided better documentation of how to use the iterator method, and the separate platform directories have different serializers that lend themselves to this

**Pull request merged**